### PR TITLE
Refactor translations

### LINKED ISSUE #1751: Way of maintaining translations up to date
# Feature Request

**Is your feature request related to a problem? Please describe.**

As a developer and user of the UI components, I would like a way to make sure that translations of geostyler are all up to date.
At least having an easy way to find missing strings in my language.
At best being warned of missing strings and being helped adding them.

**Describe the solution you'd like**

A new `npm run` command could do that by comparing translation files in any language (`src/locale/*.ts`) with the reference one in english ([src/locale/en_US.ts](https://github.com/geostyler/geostyler/blob/main/src/locale/en_US.ts)).
It would highlight the difference between files.

Suggestion : 

```npm run locale-check --language=fr_FR```
Checks `src/locale/fr_FR.ts` and warns in the console about missing strings or strings that are no longer used (not in the `en_US` file).

```npm run locale-check --language=fr_FR --fix```
Would actually modify the `src/locale/fr_FR.ts` file by adding missing strings (with the same text as in the en_US file) and delete unused strings. Only the actual translation would remain to do.

**Describe alternatives you've considered**

No other idea.

**Additional context**

There are localized strings in other repositories (such as [geostyler/geostyler-demo](https://github.com/geostyler/geostyler-demo)) and the proposal wouldn't check them.

The topic is open for discussion, there isn't any work in progress at the moment.

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

Refactors translations.

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

Closes #1751

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [BSD 2-Clause License](https://github.com/geostyler/geostyler/blob/main/)
- [x] I have followed the [guidelines for contributing](https://github.com/geostyler/geostyler/blob/main/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/geostyler/geostyler/blob/main/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Comment by jansule** [comment]
Nice idea. Will we have to extend the antd locales in the interface?

**Comment by KaiVolland** [comment]
> Nice idea. Will we have to extend the antd locales in the interface?
Done in https://github.com/geostyler/geostyler/pull/1753/commits/0ac39efdf10975129df42f33e44d31d1551bf2be

File: src/index.ts [hunk 1]
```diff
@@ -237,5 +238,6 @@ export {
   WellKnownNameEditor,
   WellKnownNameField,
   WidthField,
-  withDefaultsContext
+  withDefaultsContext,
+  GeoStylerLocale
```
Comment (jansule) [inline src/index.ts:242]: Isn't that a (non-breaking) API change?

File: src/Component/BulkEditor/BulkEditor.tsx [hunk 2]
```diff
@@ -41,20 +41,13 @@ import { KindField } from '../Symbolizer/Field/KindField/KindField';
...
-  opacityLabel: string;
-  symbolLabel: string;
-  imageFieldLabel: string;
-}
 
 // default props
 interface BulkEditorDefaultProps {
   /** Locale object containing translated text snippets */
-  locale: BulkEditorLocale;
+  locale: GeoStylerLocale['BulkEditor'];
```
Comment (jansule) [inline src/Component/BulkEditor/BulkEditor.tsx:50]: I'm not sure if i like the removal of the `Gs` prefix (`GsBulkEditor` -> `BulkEditor`). We are extending the antd locales so we might run into naming collisions.

File: src/locale/es_ES.ts [hunk 3]
```diff
@@ -26,35 +26,116 @@
...
-    graphicalEditor: 'Editor gráfico',
-    codeEditor: 'Editor de código'
-  },
-  GsBulkEditModals: {
+import antd_es_ES from 'antd/lib/locale-provider/es_ES';
+import { GeoStylerLocale } from './locale';
+
+const es_ES: GeoStylerLocale = {
+  RgbChannelField: {
+    redLabel: 'es_ES:Red band',
```
Comment (jansule) [inline src/locale/es_ES.ts:34]: Why are there `es_ES:` prefixes?

File: src/locale/fr_FR.ts [hunk 4]
```diff
@@ -26,49 +26,98 @@
...
-    graphicalEditor: 'Éditeur graphique',
-    codeEditor: 'Éditeur de code'
-  },
-  GsBulkEditModals: {
+import antd_fr_FR from 'antd/lib/locale-provider/fr_FR';
+import { GeoStylerLocale } from './locale';
+
+const fr_FR: GeoStylerLocale = {
+  StyleFieldContainer: {
+    nameFieldLabel: 'fr_FR:Name',
```
Comment (jansule) [inline src/locale/fr_FR.ts:34]: See comment above

File: src/locale/fr_FR.ts [hunk 5]
```diff
@@ -241,39 +291,42 @@ export default {
...
+  SymbolizerEditor: {
     kindFieldLabel: 'Type'
   },
-  GsSymbolizerEditorWindow: {
+  SymbolizerEditorWindow: {
     symbolizersEditor: 'Éditeur de symbolisation'
   },
-  GsFilterEditorWindow: {
+  FilterOverview: {
+    filterTitle: 'Filters'
```
Comment (jansule) [inline src/locale/fr_FR.ts:317]: Is this really French?

File: src/locale/zh_CN.ts [hunk 6]
```diff
@@ -26,78 +26,98 @@
...
-    graphicalEditor: '编辑器',
-    codeEditor: '代码编辑器'
-  },
-  GsBulkEditModals: {
+import antd_zh_CN from 'antd/lib/locale-provider/zh_CN';
+import { GeoStylerLocale } from './locale';
+
+const zh_CN: GeoStylerLocale = {
+  StyleFieldContainer: {
+    nameFieldLabel: 'zh_CN:Name',
```
Comment (jansule) [inline src/locale/zh_CN.ts:34]: See comment above

File: src/locale/fr_FR.ts (hunk 5 above)
Comment (KaiVolland) [reply src/locale/fr_FR.ts:317]: I guess sometimes its more natural to keep the english word. But i guess we're the wrong persons to discuss this. :rofl:

**Comment by KaiVolland** [comment]
The prefixes `fr_FR:` … in the translations are on purpose as these are missing translations.

File: src/Component/BulkEditor/BulkEditor.tsx (hunk 2 above)
Comment (KaiVolland) [reply src/Component/BulkEditor/BulkEditor.tsx:50]: Currently there are no conflicts... IMHO we can think about this when we run into conflicts.

File: src/locale/fr_FR.ts (hunk 5 above)
Comment (jansule) [reply src/locale/fr_FR.ts:317]: So this is a translation that was already provided?

**Comment by slafayIGN** [comment]
Nicely done !
I agree with @jansule that prefixing with `TODO:` would be a more intuitive naming.

**Review by jansule — APPROVED** [review]
Very nice @KaiVolland! Thanks for addressing the issues.

**Pull request merged**